### ISSUE TITLE
Where clause not entirely used in the SELECT query after the UPDATE

### ISSUE THREAD

**Issue by frederikhors** [ent/ent#2656]
I'm using tenant_id column with Filter rule as you suggested with that example. It is working good but now I found this issue.

When I edit a player (simply changing a field, for example) it executes these queries all in the same transaction:

```
Tx(5979fed9-b9ee-4f7d-abd8-def438da41c2).Query: query=SELECT DISTINCT "players"."id", "players"."tenant_id", "players"."created_at", "players"."updated_at", "players"."name" FROM "players" WHERE "players"."id" = $1 AND "players"."tenant_id" = $2 LIMIT 1 args=[1 3]

Tx(5979fed9-b9ee-4f7d-abd8-def438da41c2).Exec: query=UPDATE "players" SET "name" = $1 WHERE "id" = $2 AND "players"."tenant_id" = $3 args=[Bob 1 3]

Tx(5979fed9-b9ee-4f7d-abd8-def438da41c2).Query: query=SELECT "id", "tenant_id", "created_at", "updated_at", "name" FROM "players" WHERE "id" = $1 args=[1]
```

1. the first one is to get the player for business operations and it is using `tenant_id`, good

2. the second one is the UPDATE and it is using `tenant_id`, good

3. the third one is the SELECT after the UPDATE which is something that automatically Ent does. As you can see it is NOT using `tenant_id`; I am worried about that; I know you are working to remove that second query but until this change...

**Can I be sure that everything is correct?**

**Why is this query without the `tenant_id`? Doesn't it follow the same logic as the other queries?**

**Comment by a8m** [comment]
Hey @frederikhors!
This is actually a correct behavior and it's explained [here](https://github.com/ent/ent/blob/master/dialect/sql/sqlgraph/graph.go#L701-L703). 

TL;DR, in case one of the fields used by the predicate were changed by the UPDATE statement the SELECT won't return any value. Therefore, in case **there's a match** we rely only on the ID field that **identifies the matched user** as it's the only field that cannot be changed by Ent.

**Comment by frederikhors** [comment]
Ok. Thanks. I think the multi primary keys feature we'll "fix" this too.
